fail2ban/server/filter.py

```diff
@@ -855,8 +855,9 @@ def dnsToIp(dns):
 		""" Convert a DNS into an IP address using the Python socket module.
 			Thanks to Kevin Drapel.
 		"""
+		# retrieve ip (todo: use AF_INET6 for IPv6)
 		try:
-			return set(socket.gethostbyname_ex(dns)[2])
+			return set([i[4][0] for i in socket.getaddrinfo(dns, None, socket.AF_INET, 0, socket.IPPROTO_TCP)])
```

Comment:
and as regards the defaults: sometimes that will prevent to retrieve the same adresses twice (kind of a `distinct`, I know that the `set` does also, but ...) - if you don't like it, I can remove it.